e2e/test/iothub/telemetry/TelemetrySendE2ETests.cs

```diff
@@ -0,0 +1,261 @@
+            await deviceClient.CloseAsync().ConfigureAwait(false);
+        }
+
+        public static async Task SendSingleMessageAsync(DeviceClient deviceClient, string deviceId, MsTestLogger logger, int messageSize = 0)
+        {
+            Client.TelemetryMessage testMessage;
+
+            if (messageSize == 0)
+            {
+                (testMessage, _) = ComposeTelemetryMessage(logger);
+            }
+            else
+            {
+                (testMessage, _) = ComposeComplexTelemetryMessage(messageSize, logger);
+            }
+
+            using (testMessage)
+            {
+                await deviceClient.SendTelemetryAsync(testMessage).ConfigureAwait(false);
+            }
+        }
+
+        public static async Task SendSingleMessageWithComponentAsync(DeviceClient deviceClient, string deviceId, MsTestLogger logger, int messageSize = 0)
+        {
+            Client.TelemetryMessage testMessage;
+
+            if (messageSize == 0)
+            {
+                (testMessage, _) = ComposeTelemetryMessageWithComponent(logger);
+            }
+            else
+            {
+                (testMessage, _) = ComposeComplexTelemetryMessageWithComponent(messageSize, logger);
+            }
+
+            using (testMessage)
+            {
+                await deviceClient.SendTelemetryAsync(testMessage).ConfigureAwait(false);
+            }
+        }
+
+        public static (Client.TelemetryMessage message, string p1Value) ComposeTelemetryMessageWithComponent(MsTestLogger logger)
+        {
+            string messageId = Guid.NewGuid().ToString();
+            string p1Value = Guid.NewGuid().ToString();
+            string userId = Guid.NewGuid().ToString();
+            string componentName = Guid.NewGuid().ToString();
+
+            logger.Trace($"{nameof(ComposeTelemetryMessage)}: messageId='{messageId}' userId='{userId}' p1Value='{p1Value}'");
+            var message = new TelemetryMessage
+            {
+                MessageId = messageId,
+                UserId = userId,
+                ComponentName = componentName,
+                Telemetry = {
+                   { "property1", p1Value },
+                   { "property2", null},
+                }
+            };
+
+            return (message, p1Value);
+        }
+
+        public static (Client.TelemetryMessage message, string p1Value) ComposeComplexTelemetryMessage(int messageSize, MsTestLogger logger)
+        {
+            string messageId = Guid.NewGuid().ToString();
+            string p1Value = Guid.NewGuid().ToString();
+
+            logger.Trace($"{nameof(ComposeComplexTelemetryMessage)}: messageId='{messageId}' p1Value='{p1Value}'");
+            var message = new TelemetryMessage
+            {
+                MessageId = messageId,
+            };
+            message.Telemetry.Add("property1", p1Value);
+            message.Telemetry.Add("property2", null);
+            message.Telemetry.Add("property3", new
```

Comment:
do you think something like this would be better as a unit test? when we can check that we serialize and encode the message properly?